src/CustomFetch.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react';
+
+function useCustomFetch(url) {
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    setLoading(true);
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((result) => {
+        if (isMounted) {
+          setData(result);
+          setLoading(false);
+        }
+      })
+      .catch((error) => {
```

Comment:
What would this look like with `async/await` inside of a `try/catch` block?

Reply:
I've updated the code to use async/await. My understanding is that the benefit here would be that async/await results in more readable, maintainable, and manageable asynchronous code, that also offers better error handling compared to traditional promise chaining, correct?